mock/mock-functions.js

```diff
@@ -1,26 +1,54 @@
-const onFdc3Ready = () => new Promise((resolve) => {
-  if (window.fdc3) {
+const onFdc3Ready = () =>
+  new Promise((resolve) => {
+    if (window.fdc3) {
       resolve();
-  } else {
-      window.addEventListener('fdc3Ready', () => resolve());
-  }
-});
+    } else {
+      window.addEventListener("fdc3Ready", () => resolve());
+    }
+  });
 
 const closeWindowOnCompletion = async (fdc3) => {
-  const appControlChannel = await window.fdc3.getOrCreateChannel(
-      "app-control"
-    );
-   appControlChannel.addContextListener("closeWindow", async (context) => {
+  let implementationMetadata = await fdc3.getInfo();
+  let { fdc3Version } = implementationMetadata;
+  const appControlChannel = await fdc3.getOrCreateChannel("app-control");
+  if (fdc3Version === "1.2") {
+    await closeWindowOnCompletion_1_2(appControlChannel);
+  } else if (fdc3Version === "2.0") {
+    await closeWindowOnCompletion_2_0(appControlChannel);
+  }
+};
+
+const closeWindowOnCompletion_1_2 = async (appControlChannel) => {
+  appControlChannel.addContextListener("closeWindow", async (context) => {
+    //notify app A that window was closed
+    appControlChannel.broadcast({
+      type: "windowClosed",
+      testId: context.testId,
+    });
+    await sleep(10);
+    //yield to make sure the broadcast gets out before we close
+    window.close();
+  });
+};
 
-      //notify app A that window was closed
-      appControlChannel.broadcast({
-        type: "windowClosed",
-        testId: context.testId,
-      });
-      setTimeout(() => {
-        //yield to make sure the broadcast gets out before we close
-        window.close();
-        return;
-      }, 1);
+const closeWindowOnCompletion_2_0 = async (appControlChannel) => {
+  await appControlChannel.addContextListener("closeWindow", async (context) => {
+    //notify app A that window was closed
+    await appControlChannel.broadcast({
+      type: "windowClosed",
+      testId: context.testId,
+    });
+    await sleep(10);
+    //yield to make sure the broadcast gets out before we close
+    window.close();
   });
 };
+
+async function sleep(timeoutMs) {
```

Comment:
You can pull this from the utils file - along with a constant for the sleep time...

Really this file (and the scripts in the mock apps) should be typescript and should be built then linked in the head tag.

Reply:
I know, it's on a long list of todos

Reply:
> I know, it's on a long list of todos

I'll quit grumbling then ;-)

Reply:
No worries, we should have done this from the start but as I've only recently started dabbling in Typescript/node.js, I wasn't familiar with Webpack or any other Typescript-to-Javascript transpiler tool and wasn't sure how to do this at the time